### PR TITLE
Update multi scatter Ignore nan in the sum of peaks.

### PR DESCRIPTION
**Pull request by michaweiss89**
_Before you submit this PR: make sure to put all operations-related information in a wiki-note, a PR should be about code and is publicly accessible_

## What does the code in this PR do / what does it improve?
It ignores nan in the sum. 
## Can you briefly describe how it works?
It ignores nan in the sum. 
## Can you give a minimal working example (or illustrate with a figure)?
Same as before

considerations._

All _italic_ comments can be removed from this template.

### CODE REVIEW

File: straxen/plugins/events/multi_scatter.py [hunk 1]
```diff
@@ -55,36 +38,56 @@ def infer_dtype(self):
...
-            result[f'cs2_sum'][event_i] = np.sum(sp[cond]['cs2'])
-            result[f'cs2_wo_timecorr_sum'][event_i] = np.sum(sp[cond]['cs2_wo_timecorr'])
-            result[f'cs2_wo_elifecorr_sum'][event_i] = np.sum(sp[cond]['cs2_wo_elifecorr'])
-            result[f'cs2_area_fraction_sum'][event_i] = np.sum(sp[cond]['cs2_area_fraction_top'])            
-            result[f's1_sum'][event_i] = np.sum(sp[sp["type"]==1]['area'])
-            if np.sum(sp[cond]['cs2']) > 0: 
-                result[f'cs1_multi_wo_timecorr'][event_i] = event["s1_area"] * np.average(sp[cond]['s1_xyz_correction_factor'], weights = sp[cond]['cs2'])
-                result[f'cs1_multi'][event_i] = result[f'cs1_multi_wo_timecorr'][event_i] * np.average(sp[cond]['s1_rel_light_yield_correction_factor'], weights = sp[cond]['cs2'])
+            cond = (sp["type"] == 2) & (sp["drift_time"] > 0)
+            cond &= (sp["drift_time"] < 1.01 * self.drift_time_max) & (sp["cs2"] > 0)
```
Comment (dachengx) [inline straxen/plugins/events/multi_scatter.py:76]: Hey, @michaweiss89. Would you please make this `1.01` a constant configured by `URLConfig`?

**Review by dachengx — CHANGES_REQUESTED** [review]

File: straxen/plugins/peaks/peak_corrections.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+from straxen.plugins.events.corrected_areas import CorrectedAreas
+import strax
+import numpy as np
+import straxen
+
+export, __all__ = strax.exporter()
+
+
+@export
+class PeakCorrectedAreas(CorrectedAreas):
```
Comment (dachengx) [inline straxen/plugins/peaks/peak_corrections.py:10]: Would you please add doctoring to this plugin `PeakCorrectedAreas`?

File: straxen/plugins/peaks/peak_corrections.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    __version__ = '0.0.1'
+
+    depends_on = ('peak_basics', 'peak_positions', 'peak_per_event')
+    data_kind = 'peaks'
+    provides = 'peak_corrections'
```
Comment (dachengx) [inline straxen/plugins/peaks/peak_corrections.py:15]: Also, do you mind if I change this to `peak_corrections`? Usually, we need to use `peak` in plugins except `peaks`.

File: straxen/plugins/peaks/peak_corrections.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    def compute(self, peaks):
+        result = np.zeros(len(peaks), self.dtype)
+        result['time'] = peaks['time']
+        result['endtime'] = peaks['endtime']
+
+        # Get z position of the peak
+        z_obs = -self.electron_drift_velocity * peaks['drift_time']
+        z_obs = z_obs + self.electron_drift_velocity * self.electron_drift_time_gate
+        result['z_obs_ms'] = z_obs
```
Comment (dachengx) [inline straxen/plugins/peaks/peak_corrections.py:62]: Just from curiosity, why we did not apply field-distortion correction here?

File: straxen/plugins/events/multi_scatter.py [hunk 5]
```diff
@@ -55,37 +39,71 @@ def infer_dtype(self):
...
-             'top of cathode wires [cm]', )
+             'top of cathode wires [cm]')
+
+    ms_window_fac = straxen.URLConfig(
+        default=1.01, type=(int, float),
+        help='Max drift time window to look for peaks in multiple scatter events'
+    )
+
+    def infer_dtype(self):
+        dtype = strax.time_fields + [
```
Comment (dachengx) [inline straxen/plugins/events/multi_scatter.py:50]: Would you also add comments on these variables?

File: straxen/plugins/events/multi_scatter.py [hunk 6]
```diff
@@ -3,47 +3,31 @@
...
-            ((f'e_charge_sum'), np.float32),
-             ]
-        dtype += strax.time_fields
-        return dtype
-
+    __version__ = '0.0.2'
+    depends_on = (
+        'event_info',
+        'peak_basics', 'peak_per_event', 'peak_corrections', 'peak_positions')
+    provides = 'event_ms_naive'
```
Comment (dachengx) [inline straxen/plugins/events/multi_scatter.py:16]: Do you mind if I change this to `event_ms_naive`? It is better to use lowercase in `provides`.

**Review by dachengx — CHANGES_REQUESTED** [review]

File: straxen/plugins/events/multi_scatter.py (hunk 5 above)
Comment (HenningSE) [reply straxen/plugins/events/multi_scatter.py:50]: Done in https://github.com/XENONnT/straxen/pull/1162/commits/ec47dd20dbf7fec29ae742f5672f7345410c0276

**Comment by dachengx** [comment]
Or I guess we can also implement the FDC-corrected z later.

**Comment by dachengx** [comment]
@HenningSE Do you think this PR is in good shape to merge? Or what are the ongoing or pending development?

**Review by dachengx — APPROVED** [review]

**Comment by HenningSE** [comment]
Hi @dachengx, sorry I think I missed your question a few days ago. As far as I can tell, the FDC correction was basically copy and paste from the existing straxen code. 

From my perspective the PR can be merged! Thanks for you help!

**Comment by dachengx** [comment]
> Hi @dachengx, sorry I think I missed your question a few days ago. As far as I can tell, the FDC correction was basically copy and paste from the existing straxen code.
> 
> From my perspective the PR can be merged! Thanks for you help!

Hey @HenningSE . The `z_obs_ms` in `peak_corrections.py` is only corrected by `electron_drift_time_gate`. So the FDC correction is not done. But anyway we can do it later and I believe this PR was meant to fix bugs. 

I will merge it now.

**Pull request merged**